packages/resolvers/glob/src/GlobResolver.js

```diff
@@ -0,0 +1,125 @@
+// @flow
+import {Resolver} from '@parcel/plugin';
+import {isGlob, glob, relativePath, normalizeSeparators} from '@parcel/utils';
+import micromatch from 'micromatch';
+import path from 'path';
+import nullthrows from 'nullthrows';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+
+export default (new Resolver({
+  async resolve({dependency, options, filePath, pipeline}) {
+    if (!isGlob(filePath)) {
+      return;
+    }
+
+    let sourceAssetType = nullthrows(dependency.sourceAssetType);
+    let sourceFile = nullthrows(
+      dependency.resolveFrom ?? dependency.sourcePath,
```

Comment:
I guess we should change the return type of `dependency.resolveFrom` to `FilePath`?

https://github.com/parcel-bundler/parcel/blob/98c5902cefe1bba6c890da56f111e4b31b487c64/packages/core/core/src/public/Dependency.js#L111-L113